rdkPlugins/LocalTime/source/LocalTimePlugin.cpp

```diff
@@ -83,6 +85,54 @@ bool LocalTimePlugin::postInstallation()
     return true;
 }
 
+// -----------------------------------------------------------------------------
+/**
+ *  @brief postInstallation OCI hook.
+ *
+ *  If set_tz parameter is set then its value should be a path to file.
+ *  Read this file and put its contents into containers TZ env var.
+ *
+ *  @return true on success, false on failure.
+ */
+bool LocalTimePlugin::preCreation()
+{
+    AI_LOG_FN_ENTRY();
+
+    if (!mContainerConfig)
+    {
+        AI_LOG_WARN("container config is null");
+        return false;
+    }
+
+    const char* set_tz = mContainerConfig->rdk_plugins->localtime->data->set_tz;
+    if (set_tz)
+    {
+        AI_LOG_DEBUG("set_tz is '%s'", set_tz);
+
+        std::ifstream file(set_tz);
+        if (file.is_open())
+        {
+            std::string tz;
+            std::getline(file, tz);
+            mUtils->addEnvironmentVar("TZ=" + tz);
+
+            AI_LOG_DEBUG("read from set_tz: %s", tz.c_str());
+        }
+        else
+        {
+            AI_LOG_WARN("unable to open '%s'", set_tz);
```

Comment:
This should be `AI_LOG_ERR` as you are failing, but as this is the only wrong thing I would not make a commit for fixing just this one thing